src/protocolgame.cpp

```diff
@@ -3570,8 +3570,8 @@ void ProtocolGame::AddShopItem(NetworkMessage& msg, const ShopInfo& item)
 
 	msg.addString(item.realName);
 	msg.add<uint32_t>(it.weight);
-	msg.add<uint32_t>(item.buyPrice == std::numeric_limits<uint32_t>::max() ? 0 : item.buyPrice);
-	msg.add<uint32_t>(item.sellPrice == std::numeric_limits<uint32_t>::max() ? 0 : item.sellPrice);
+	msg.add<uint32_t>(std::max<uint32_t>(item.buyPrice, 0));
+	msg.add<uint32_t>(std::max<uint32_t>(item.sellPrice, 0));
```

Comment:
This does not work as you think it does. This first converts both values to the template type (uint32_t). Since the buyPrice/sellPrice can be -1, this will just overflow and return that value instead of 0.

Reply:
Then just use int32 instead of u32

Reply:
> This does not work as you think it does. This first converts both values to the template type (uint32_t). Since the buyPrice/sellPrice can be -1, this will just overflow and return that value instead of 0.
![GIF 19-07-2022 02-47-55 p  m](https://user-images.githubusercontent.com/28090948/179826123-d4f3e5d1-2613-4a8f-a429-aba201a13ec1.gif)

Confirmed!
I guess a more explicit check should be done since buyPrice and sellPrice are of type `int64_t`

A possible solution:
```cpp
std::max<int64_t>(std::min<int64_t>(number, std::numeric_limits<uint32_t>::max()), 0);
```